scripts/stackstorm.js

```diff
@@ -237,6 +251,9 @@ module.exports = function(robot) {
       .catch(function (err) {
         var error_msg = 'Failed to retrieve commands from "%s": %s';
         robot.logger.error(util.format(error_msg, env.ST2_API_URL, err.message));
+        if (err.status === 401 && err.message.includes("Unauthorized")) {
```

Comment:
What if by refactoring reasons backend returned `401` HTTP code and `Authorization Required` as a message?
I think `||` would be more appropriate here.

Reply:
It doesn't make sense to handle an unauthorized error differently than another non-connection error, so I had it always handle it this way. The current code looks like (`function loadCommands`):

```js
      return (...).then(function () {
        ...
      })
      .catch(function (err) {
        var error_msg = 'Failed to retrieve commands from "%s":\n%s';
        err.stack = util.format(error_msg, env.ST2_API_URL, err.stack);
        robot.logger.error(err.stack);
        throw err;
```

This allows the `loadCommands` promise to log the error before it rejects, and then rejects are either ignored (`function start`):

```js
      // Add an interval which tries to re-load the commands
      commands_load_interval = setInterval(
        // Lambda function that swallows errors so they don't propagate up the stack
        // (errors are still logged within loadCommands itself)
        function () {
          loadCommands().catch(function (err) {});
        },
        (env.ST2_COMMANDS_RELOAD_INTERVAL * 1000));
```

or the rejection is simply passed up the stack (`function start`):
```js
      // Initial command loading
      return loadCommands();
```

and handled elsewhere (init code):

```
  // Authenticate with StackStorm backend and then call start.
  return authenticated.then(function () {
    return start().then(function () {
      return stop;
    });
  }).catch(function (err) {
    // Since start is only called once when the plugin is loaded, bail
    // loudly and harshly if loadCommands() isn't successful
    stop({shutdown: true});
    throw err;
  });
```